nodejs/awsx/apigateway/apikey.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These APIs are currently experimental and may change.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+import * as api from "./api";
+import { SecurityDefinition } from "./swagger_json";
+
+/**
+ * Input properties for creating a UsagePlan and associated API Keys.
+ */
+export interface APIKeyArgs {
+
+    /**
+     * Define the apis you would like to associate the usage plan with. This can be used in place of
+     * defining the apiStages defined in the [usagePlan]. You cannot define both [apis] and
+     * [usagePlan.apiStages].
+     */
+    apis?: api.API[];
+
+    /**
+     * Define the usage plan to create. You can either define:
+     *  1 - an existing Usage Plan - the API Keys will be associated with the usage plan
+     *  2 - UsagePlanArgs with [usagePlan.apiStages] defined to define a new Usage Plan
+     *  3 - UsagePlanArgs with [apis] defined and [usagePlan.apiStages] NOT defined to define a new
+     *      Usage Plan
+     */
+    usagePlan?: aws.apigateway.UsagePlan | aws.apigateway.UsagePlanArgs;
+
+    /**
+     * The API keys you would like to create & associate with the usage plan. You can pass an array
+     *  that has a combination of:
+     *  1 - an existing APIKey
+     *  2 - ApiKeyArgs for a new APIKey
+     */
+    apiKeys?: Array<aws.apigateway.ApiKey | aws.apigateway.ApiKeyArgs>;
+}
+
+export interface AssociatedAPIKeys {
+    readonly usagePlan: aws.apigateway.UsagePlan;
+    readonly keys: Key[];
+}
+
+export interface Key {
+    apikey: aws.apigateway.ApiKey;
+    usagePlanKey: aws.apigateway.UsagePlanKey;
+}
```

Comment:
on the fence if these two exported interfaces should be doc'ed.  They're both nearly clear enough to be ok.  If you feel like they would benefit from an example in the docs, then doc'ing would be good.  However, if the docs are merely going to be boilerplate like "the usage place this api key is associated with", then we don't really need anything here.